### ISSUE TITLE
Compiled functions can't take variable number of arguments or use keyword-only arguments with defaults

### ISSUE THREAD

**Issue by KevinHooah** [pyg-team/pytorch_geometric#9638]
## Bug Description
The latest release mentions fixing the issue of converting the model to TorchScript when it contains message_passing. However, we tested it and found that this bug remains.

#### The error message:

> Compiled functions can't take variable number of arguments or use keyword-only arguments with defaults:
>   File "some_path/pytorch_geometric-master/torch_geometric/nn/conv/message_passing.py", line 425
>         edge_index: Adj,
>         size: Size = None,
>         **kwargs: Any,
>          ~~~~~~~ <--- HERE
>     ) -> Tensor:
>         r"""The initial call to start propagating messages.

#### The toy model taken from [official tutorial](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_gnn.html):
```python 
import torch
import torch.nn.functional as F
from torch_geometric.nn import GCNConv
from torch_geometric.nn import MessagePassing
from torch.nn import Linear, Parameter
from torch_geometric.utils import add_self_loops, degree

class GCNConv(MessagePassing):
    def __init__(self, in_channels, out_channels):
        super().__init__(aggr='add') 
        self.lin = Linear(in_channels, out_channels, bias=False)
        self.bias = Parameter(torch.empty(out_channels))
        self.reset_parameters()
    def reset_parameters(self):
        self.lin.reset_parameters()
        self.bias.data.zero_()
    def forward(self, x, edge_index):
        edge_index, _ = add_self_loops(edge_index, num_nodes=x.size(0))
        x = self.lin(x)
        row, col = torch.split(edge_index, 1, dim=0) #<-- We modified this line
        deg = degree(col, x.size(0), dtype=x.dtype)
        deg_inv_sqrt = deg.pow(-0.5)
        deg_inv_sqrt[deg_inv_sqrt == float('inf')] = 0
        norm = deg_inv_sqrt[row] * deg_inv_sqrt[col]
        out = self.propagate(edge_index, x=x, norm=norm)
        out = out + self.bias
        return out
    def message(self, x_j, norm):
        return norm.view(-1, 1) * x_j

test_model = GCNConv(10, 10)
```
#### The line triggered the issue:
`script_test_model = torch.jit.script(test_model)`

### A side bug we found during this process:
If we completely follow the official tutorial's code without modifying that line with a comment, i.e.,
```python 
# the original code in the tutorial
row, col = edge_index
```
The error message will be:

> Tensor cannot be used as a tuple:
>   File "/var/folders/3f/5776h3152rs7rlmtldx8gyxh0000gn/T/ipykernel_15097/3913400558.py", line 14
>         x = self.lin(x)
>         # row, col = torch.split(edge_index, 1, dim=0)
>         row, col = edge_index
>                    ~~~~~~~~~~ <--- HERE
>         deg = degree(col, x.size(0), dtype=x.dtype)
>         deg_inv_sqrt = deg.pow(-0.5)

### Versions

I am testing it with my old mac, but this issue persists when we test our model on the GPU machine.

Versions of relevant libraries:
[pip3] numpy==1.22.4
[pip3] torch==2.2.1
[pip3] torch-cluster==1.6.0
[pip3] torch-geometric==2.2.0
[pip3] torch-scatter==2.1.0
[pip3] torch-sparse==0.6.16
[pip3] torch-spline-conv==1.2.1
[pip3] torchaudio==0.12.1
[pip3] torchvision==0.13.1
[conda] blas                      1.0                         mkl  
[conda] ffmpeg                    4.3                  h0a44026_0    pytorch
[conda] mkl                       2021.4.0           hecd8cb5_637  
[conda] mkl-service               2.4.0           py310hca72f7f_0  
[conda] mkl_fft                   1.3.1           py310hf879493_0  
[conda] mkl_random                1.2.2           py310hc081a56_0  
[conda] numpy                     1.22.4                   pypi_0    pypi
[conda] torch                     2.2.1                    pypi_0    pypi
[conda] torch-cluster             1.6.0                    pypi_0    pypi
[conda] torch-geometric           2.2.0                    pypi_0    pypi
[conda] torch-scatter             2.1.0                    pypi_0    pypi
[conda] torch-sparse              0.6.16                   pypi_0    pypi
[conda] torch-spline-conv         1.2.1                    pypi_0    pypi
[conda] torchaudio                0.12.1                py310_cpu    pytorch
[conda] torchvision               0.13.1                py310_cpu    pytorch

**Comment by rachitk** [comment]
The version of `torch_geometric` implied by your environment is 2.2.0, rather than the most recent release of 2.5.3. Does the issue persist in the most recent version of `torch_geometric`?

**Comment by KevinHooah** [comment]
> The version of `torch_geometric` implied by your environment is 2.2.0, rather than the most recent release of 2.5.3. Does the issue persist in the most recent version of `torch_geometric`?

Hello, thank you for the reply. I found I copied the environment information from the wrong one. 

We tried two approaches:
- Installing 2.5.3 from Pip
- Directly download the latest code base, and directly import all necessary codes from this local source code (in an environment where torch_geometric was not installed)

Both methods don't work, all tell me this error.

**Comment by rusty1s** [comment]
It works for me. Can you do me a favor and modify [this line](https://github.com/pyg-team/pytorch_geometric/blob/2.5.3/torch_geometric/nn/conv/message_passing.py#L196) to
```python
except Exception as e:
    print(e)
```
and see what comes back?

**Comment by KevinHooah** [comment]
> It works for me. Can you do me a favor and modify [this line](https://github.com/pyg-team/pytorch_geometric/blob/2.5.3/torch_geometric/nn/conv/message_passing.py#L196) to
> 
> ```python
> except Exception as e:
>     print(e)
> ```
> 
> and see what comes back?

Hi, following the step, the exception I got when I initiated the `GCNConv(MessagePassing)` is:
`<class '__main__.GCNConv'> is a built-in class`

**Comment by KevinHooah** [comment]
Hello, a little more info: even though I changed the model's name to GCNConv_test(MessagePassing), the exception remains the same.

**Comment by akihironitta** [comment]
Closing this issue because TorchScript has been unmaintained by the PyTorch team for 3+ years, and PyTorch 2.9 has officially deprecated its support in favor of PT2 stack.
